schema/in_memory.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/conduitio/conduit-commons/schema"
+	"github.com/conduitio/conduit-connector-protocol/pconduit"
+)
+
+var ErrSchemaNotFound = errors.New("schema not found")
+
+type inMemoryService struct {
+	// schemas is a map of schema subjects to all the versions of that schema
+	// versioning starts at 1, newer versions are appended to the end of the versions slice.
+	schemas map[string][]schema.Schema
+	// m guards access to schemas
+	m sync.Mutex
+}
+
+func NewInMemoryService() pconduit.SchemaService {
```

Comment:
Can we make the constructor private? Seems like we don't need to expose it.

Reply:
@lovromazgon It's actually being used on Conduit https://github.com/ConduitIO/conduit/blob/schema-support/pkg/conduit/runtime.go#L155-L158.

Reply:
That's not the same one, that's pointing to the `schemaregistry` package in Conduit.

Reply:
@lovromazgon you're right 🤦‍♂️